### PR TITLE
Feature/rgb image scaling

### PR DESCRIPTION
**Pull request by jnation3406**
Added an extra page to get scaling parameters for input images in the OperationWizard. Also made adding the operation contingent on setting the required number of parameters in the wizard. I also attempted to fix some of the random errors we would get when caching jpegs that couldn't be found.

### CODE REVIEW

File: public/drawImageWorker.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
+
```
Comment (jnation3406) [inline public/drawImageWorker.js:1]: Web workers need to be somewhere accessible for the frontend main thread to access them during exection, so this public dir works.

File: src/utils/common.js [hunk 2]
```diff
@@ -3,4 +3,28 @@ const calculateColumnSpan = (imageCount, imagesPerRow) => {
   return totalColumns
 }
 
-export { calculateColumnSpan }
+const convertFilter = (filter) => {
+  if (filter === 'V' || filter === 'gp') {
```
Comment (jnation3406) [inline src/utils/common.js:7]: This function maps the potential filters into 'r', 'g', 'b'. We can expand this as we accept more filter names to mean those colors.

File: vue.config.js [hunk 3]
```diff
@@ -6,5 +6,11 @@ module.exports = defineConfig({
 		vuetify: {
 			// https://github.com/vuetifyjs/vuetify-loader/tree/next/packages/vuetify-loader
 		}
+	},
+	devServer: {
```
Comment (jnation3406) [inline vue.config.js:10]: This is necessary for your locally running devServer to be able to use web workers with shared data arrays. It requires https and same origin normally, so I think it should work in production but we may need to set some cors thing for that as well.

File: src/components/Global/ImageScaler.vue [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+      rawData.value = response
+      zScaleValues = [response.zmin, response.zmax]
+      zScaleImage()
+      isLoading.value = false
+    },
+    failCallback: (error) => {
+      isLoading.value = false
+      errorReason.value = error
+      console.error('API call failed with error:', error)
+    }
```
Comment (LTDakin) [inline src/components/Global/ImageScaler.vue:78]: this is a very readable way to use lamdas for the fetchapi method, I'll be stealing this

**Review by LTDakin — COMMENTED** [review]
I gave it a look through and it looks really good, I think @mgdaily should give the final approval since he initially prototyped this

File: src/components/Global/ImageScaler.vue [hunk 5]
```diff
@@ -0,0 +1,150 @@
...
+  }
+})
+
+const emit = defineEmits(['updateScaling'])
+const store = useConfigurationStore()
+const dataSessionsUrl = store.datalabApiBaseUrl
+const isLoading = ref(true)
+const errorReason = ref('')
+const rawData = ref({})
+const sliderRange = ref([0, 65535])
```
Comment (mgdaily) [inline src/components/Global/ImageScaler.vue:35]: Maybe we add a TODO to make this work with an arbitrary bit depth? The site "simple-stacked" images will be 32 bit.

File: src/components/Global/ImageScaler.vue [hunk 6]
```diff
@@ -0,0 +1,150 @@
...
+        :image-name="props.imageName"
+        :composite-name="props.compositeName"
+      >
+      </raw-scaled-image>
+    </v-row>
+    <v-row>
+      <v-range-slider
+        v-model="sliderRange"
+        step="500"
+        max="65535"
```
Comment (mgdaily) [inline src/components/Global/ImageScaler.vue:102]: Again, may want to make this computed based on the bit depth of the image.

**Review by mgdaily — APPROVED** [review]
Mostly looks good! Just some questions about bit depth.

File: src/components/Global/ImageScaler.vue (hunk 5 above)
Comment (jnation3406) [reply src/components/Global/ImageScaler.vue:35]: hooked in the bitdepth

**Pull request merged**